pkg/providers/v1/aws.go

```diff
@@ -1503,6 +1503,23 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 			}
 		}
 
+		for _, macID := range macIDs {
+			ipv6Path := path.Join("network/interfaces/macs/", macID, "ipv6s")
+			internalIPv6s, err := c.metadata.GetMetadata(ipv6Path)
+			if err != nil {
+				// Interfaces without an IPv6 address don't get the "ipv6s" data
+				klog.V(4).Infof("error querying AWS metadata for %q: %q", ipv6Path, err)
+				continue
+			}
+
+			for _, internalIPv6 := range strings.Split(internalIPv6s, "\n") {
+				if internalIPv6 == "" {
+					continue
+				}
+				addresses = append(addresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: internalIPv6})
```

Comment:
Should we should use it only if ip.IsGlobalUnicast()

Reply:
Why should it exclude site-local unicast?

Reply:
they are not routable, so other nodes will not be able to reach the node on that address

Reply:
ah, wait , you mean site-local , those were deprecated on RFC 3879, there no site-local since then.

AFAIK https://golang.org/pkg/net/#IP.IsGlobalUnicast covers all the routable addresses.

Reply:
Didn't add any check because wanted to keep it as close as possible to the IPv4 handling.
Maybe this would be something to do in a follow-up PR and try to improve both.

Reply:
yeah, keeping it consistent looks better to me, you can discard this comment

Reply:
👍